src/library_glfw.js

```diff
@@ -671,6 +672,64 @@ var LibraryGLFW = {
       win.scrollFunc = cbfun;
     },
 
+    setDropCallback: function(winid, cbfun) {
+      var win = GLFW.WindowFromId(winid);
+      if (!win) return;
+      win.dropFunc = cbfun;
+    },
+
+    onDrop: function(event) {
+      if (!GLFW.active.dropFunc) return;
+
+      event.preventDefault();
+
+      var filenames = allocate(new Array(event.dataTransfer.files.length*4), 'i8*', ALLOC_NORMAL);
+      var filenamesArray = [];
+      var count = event.dataTransfer.files.length;
+
+      // Read and save the files to emscripten's FS
+      // TODO: lazily load?
+      var written = 0;
+      for (var i = 0; i < count; ++i) {
+        (function(file) {
+          var reader = new FileReader();
+          reader.onload = function(e) {
+            var data = e.target.result;
+            var path = file.name; // TODO: to a new directory?
+            FS.writeFile(path, new Uint8Array(data), { encoding: 'binary' });
```

Comment:
The behavior with dropping files on the web will differ quite a bit from that of dropping files in a native GLFW application, in that in a native app, the file already exists on the filesystem and no new files are created, but in here, we create a new file to the filesystem. This file is not cleaned up anywhere, so it is up to the receiver of the file to `unlink()` all the files it receives in the handler (especially if it is not interested in any of them). Is this desired behavior? If so, it would be good to document this behavioral difference, since that can be a big source of memory leaks since GLFW applications will need to know about this new `unlink()` responsibility for all undesired files.

Alternatively, if it would be better that this handler would `FS.unlink()` all the files it adds, then that's probably also good to document that GLFW applications know not to depend on the files being present outside the drop handler. Which behavior do you think makes more sense?